lib/src/commit.rs

```diff
@@ -126,12 +126,7 @@ impl Commit {
     /// Returns whether commit's content is empty. Commit description is not
     /// taken into consideration.
     pub fn is_empty(&self, repo: &dyn Repo) -> BackendResult<bool> {
-        let parents: Vec<_> = self.parents().try_collect()?;
-        if let [parent] = &parents[..] {
-            return Ok(parent.tree_id() == self.tree_id());
-        }
-        let parent_tree = merge_commit_trees(repo, &parents)?;
-        Ok(*self.tree_id() == parent_tree.id())
+        Ok(self.tree_id() == &self.parent_tree(repo)?.id())
```

Comment:
still `parent.tree_id()` is cheaper than `parent.tree().id()`.

Reply:
Because the tree itself is not loaded, goot point. I'll retire this PR.